### PR TITLE
V1.1 branch

### PR DESCRIPTION
**Pull request by DrTimothyAldenDavis**
Dec 30, 2023: version 1.1.0

    * major change to build system: by Markus Mützel
    * port: to 32-bit systems

### CODE REVIEW

**Comment by DrTimothyAldenDavis** [comment]
I have a few more changes to add to this PR, based on feedback from the SuiteSparse 7.4.0.beta1 release.  I'm testing them now and will add them to this PR shortly.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 name: LAGraph CI
 
-on: [push, pull_request]
+on:
+  workflow_dispatch:
+  push:
+    branches-ignore:
+      - '**/*dev2'
```
Comment (michelp) [inline .github/workflows/build.yml:7]: Why this ignore?

File: ChangeLog
Comment (michelp) [inline ChangeLog:1]: Thank you Markus Mützel!

File: src/test/test_minmax.c [hunk 2]
```diff
@@ -80,7 +80,7 @@ const matrix_info files [ ] =
     {                  1,                  1, "matrix_fp32_structure.mtx" } ,
     {          -INFINITY,           INFINITY, "matrix_fp64.mtx" } ,
     {             -32768,              32767, "matrix_int16.mtx" } ,
-    {        -2147483648,         2147483647, "matrix_int32.mtx" } ,
+    {      -2147483648.0,         2147483647, "matrix_int32.mtx" } ,
```
Comment (michelp) [inline src/test/test_minmax.c:83]: Why the decimal?

**Review by michelp — COMMENTED** [review]
Overall looks great, Lots of missing frees!  @DrTimothyAldenDavis the Boehm garbage collection library has a [leak detection](https://hboehm.info/gc/leak.html) maybe it makes sense to try and interface that (maybe at the SS level) as another sanity check.

File: .github/workflows/build.yml (hunk 1 above)
Comment (DrTimothyAldenDavis) [reply .github/workflows/build.yml:7]: Because I use the dev2 branch to push to directly, and I don't want to run tests all the time.  If I want to run the test automatically, I can do that with a PR to another branch, say '*dev'.

File: src/test/test_minmax.c (hunk 2 above)
Comment (DrTimothyAldenDavis) [reply src/test/test_minmax.c:83]: Because otherwise, a compiler complains about the int32 value getting truncated.  It's safer this way and it silences the compiler warning.

**Review by michelp — APPROVED** [review]

**Comment by DrTimothyAldenDavis** [comment]
> Overall looks great, Lots of missing frees! @DrTimothyAldenDavis the Boehm garbage collection library has a [leak detection](https://hboehm.info/gc/leak.html) maybe it makes sense to try and interface that (maybe at the SS level) as another sanity check.

The src/algorithms all have "brutal" tests, which have a "brutal malloc".  The brutal malloc is given a count:  you have (say) 5 mallocs.  Count down to zero, and then after that always return NULL.  Then in the brutal tests, I give the algorithm 0 mallocs, and check the result, then 1, then 2, etc, until it succeeds.  The brutal malloc also counts malloc's vs free's, and then the test fails if the mallocs minus frees is not zero.  The LAGraph_Free also passes the size of the block being freed, so if the count of bytes still malloc'ed is not zero, then the test fails.

This 'brutal' process is thus able to detect leaks that occur when running out of memory.  That is, the method should check if it gets a NULL from malloc, and if it fails that way it should free any temporary workspace, or the result if that's been allocated.

So, there were no leaks in the src/utility or src/algorithms.  Not even any leaks in the src/tests.

Still, the brutal tests could simultaneously use a valgrind option, or other checker.

The experimental algorithms, utilities, and tests typically don't have brutal tests.  I had a segfault so I ran all the tests, including experimental, using valgrind, and that's when I found all those leaks.  So I patched them.  I think that now both the src and experimental algos, utilities, and tests are all leak free.

Still, a leak checker would be nice, particularly if it works on Windows and Mac (valgrind doesn't work on Windows, I think.  Nor on Mac).

**Comment by DrTimothyAldenDavis** [comment]
Regarding the brutal memory tests:  I could also add my own memory table for debugging only.  I do this in GraphBLAS:  each malloc gets logged in its table, with its size.  A free passes in the size, too (not just the pointer) and the size much match exactly or an error is thrown.  Then I can also check if the pointer passed to free is valid (it must be in the table).  Then when GrB_Finalize is called, it checks to ensure the table is empty, and throws an error ("leak!") if it's not empty.

It would not be hard to add this to LAGraph too, since all LAGraph_malloc's go through the same malloc function that GraphbLAS uses.

**Comment by DrTimothyAldenDavis** [comment]
Also: the *dev2 branches are not tested with the CI because they sometimes depend on GraphBLAS versions that are in the CI yet.  Some branches (v1.2_branch in particular) requires the GrB get/set, which is in the draft SuiteSparse:GraphBLAS 9.0.x.

The build.yaml file only tests with GraphBLAS 7.x, so no JIT, no GrB_get/set, and so on.

Once the v2.1 C API becomes stable, and GraphBLAS 8.2.x, 8.3.x, and 9.0.x can be added to the build.yml file, then it would be possible to enable the CI on more branches.

File: .github/workflows/build.yml (hunk 1 above)
Comment (DrTimothyAldenDavis) [reply .github/workflows/build.yml:7]: Also, *dev2 branches can contain LAGraph versions that rely on GrB get/set, which are not in the CI and they break.

File: config/LAGraph.pc.in [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# SPDX-License-Identifier: BSD-2-Clause
+
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: LAGraph
+URL: https://github.com/DrTimothyAldenDavis/SuiteSparse
+Description: Library plus test harness for collecting algorithms that use GraphBLAS in SuiteSparse
```
Comment (mmuetzel) [inline config/LAGraph.pc.in]: Should references to SuiteSparse be removed when upstreaming this to here?
``` suggestion
URL: https://github.com/GraphBLAS/LAGraph
Description: Library plus test harness for collecting algorithms that use GraphBLAS
```

Comment (DrTimothyAldenDavis) [reply config/LAGraph.pc.in]: Good point.  I'd like to keep the LAGraph repo and the SuiteSparse folder the same if possible though. So maybe both UTLs would be better.

Comment (mmuetzel) [reply config/LAGraph.pc.in]: IIUC, the pkg-config format doesn't allow two URLs:
https://manpages.ubuntu.com/manpages/jammy/man5/pc.5.html
>      URL     A URL to a webpage for the package.  This is used to recommend where newer versions
>              of the package can be acquired.  (mandatory; literal)

Comment (DrTimothyAldenDavis) [reply config/LAGraph.pc.in]: Ok. Then it should be changed and also in suitesparse to match.

Comment (DrTimothyAldenDavis) [reply config/LAGraph.pc.in]: I also added it to the dev2 branch of SuiteSparse.

File: config/LAGraph.pc.in [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# LAGraph, Copyright (c) 2019-2023, Timothy A. Davis.
```
Comment (DrTimothyAldenDavis) [inline config/LAGraph.pc.in]: fixme: wrong copyright

File: src/algorithm/LG_CC_FastSV6.c [hunk 5]
```diff
@@ -551,9 +556,9 @@ int LG_CC_FastSV6           // SuiteSparse:GraphBLAS method, with GxB extensions
         bool T_jumbled, T_iso ;
         GRB_TRY (GxB_Matrix_unpack_CSR (T, &Tp, &Tj, &Tx, &Tp_size, &Tj_size,
             &Tx_size, &T_iso, &T_jumbled, NULL)) ;
-
+// printf ("did unpack T\n") ; fflush (stdout) ; fflush (stderr) ;
```
Comment (DrTimothyAldenDavis) [inline src/algorithm/LG_CC_FastSV6.c]: delete me
